### DISCUSSION TITLE
can i use HelmRelease in kustomize.sourceRef?

### DISCUSSION THREAD

**Discussion by wibed** [fluxcd/flux2#4144] [Q&A] [answered]
the documentation says otherwise

https://fluxcd.io/flux/components/kustomize/kustomization/#source-reference

**Comment by hiddeco** [comment, marked as answer]
Not possible, as documented.

**Reply by wibed** [reply]
is it on the roadmap somewhere in the future, or completely off the table?

**Reply by hiddeco** [reply]
Why would you want this to be possible? The Kustomization has nothing to do with a HelmRelease object, as they both have their own logic around how things should be applied to a cluster.

**Reply by wibed** [reply]
i was thinking about the patch functionality.
or custom health checks.

but i remember i can put a kubernetes/kustomize resource on top that applies the patches just aswell.
id just have to add a flux/kustomize resource on top of the kubernetes/kustomize resource to apply custom health check afaik.

**Reply by hiddeco** [reply]
helm-controller itself has built-in patching: https://fluxcd.io/flux/components/helm/helmreleases/#post-renderers

For the health checking, you can do that from within the Kustomization which applies the HelmRelease.

**Reply by wibed** [reply]
thank you!
